### DISCUSSION TITLE
WalletConnectLegacy seems to be used even when `projectId` is provided

### DISCUSSION THREAD

**Discussion by fionnachan** [rainbow-me/rainbowkit#1229] [Q&A] [answered]
<img width="434" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/13184582/6dec9e8b-b0c8-4957-b42d-463c40f2456d">
this is with rainbowkit 0.12.13 and wagmi 0.12.9

I tried with the same wallet and the `WalletConnectConnector` from wagmi 0.12.9 without rainbowkit and the result is different:

<img width="343" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/13184582/744601da-1b38-4e4c-972a-145fbde95537">

Can you please let me know how to verify WC v2 instead of WC v1 is in use with rainbowkit?

**Comment by dev-cz** [comment]
I see this same issue as well. v2 is not being used when leveraged through RainbowKit.

**Comment by fionnachan** [comment]
Hi @DanielSinclair would you be able to share how to configure it to WC v2?

**Reply by DanielSinclair** [reply, marked as answer]
@fionnachan @dev-cz This is currently the expected behavior. As long as you adopt the `projectId` described [here](https://www.rainbowkit.com/docs/installation#configure), then you're all set once the WalletConnect v2 patch releases appear over the next few weeks. We'll communicate in the release notes when wallets are upgraded. Each will be upgraded individually as the level of support is still unfortunately unequal.

**Reply by dev-cz** [reply]
Ahh ok. I didn not realize that. We are using the projectId already, but with wagmi natively, the new v2 appears. So did not realize Rainbow Kit hadn't fully transitioned yet. Great, thank you!

**Reply by fionnachan** [reply]
Hi @DanielSinclair, as it's only 1 month away from WC v1's deprecation date, just want to confirm rainbowkit will make a release to pre-v1 (wagmi v1 / viem) support for WC v2, as our team will have to plan our refactoring accordingly if WC v2 won't be available with pre v1 rainbowkit

**Reply by DanielSinclair** [reply]
@fionnachan Yes, we will still offer a v2 upgrade for 0.12.x. However, we have found a few issues with WalletConnect v2 that the Wagmi and WalletConnect teams have worked on since the Wagmi v1 release, so I would (unfortunately) recommend that you make the leap so that you can easily pull in these changes. If you use ethers, you can actually adopt both viem and ethers together, so it may be a simpler upgrade.

**Reply by yuripave** [reply]
Hi @DanielSinclair, would like to know more about what are the few issues found for 0.12.x on v2 upgrade, thanks!
